pymongoexplain/commands.py

```diff
@@ -44,16 +49,40 @@ def get_SON(self):
 
 class UpdateCommand(BaseCommand):
     def __init__(self, collection: Collection, filter, update,
-                 kwargs):
-        super().__init__(collection.name)
-        return_document = {"updates":[{"q": filter, "u": update}]}
-        for key in kwargs:
-            value = kwargs[key]
-            if key == "bypass_document_validation":
-                return_document[key] = value
-            else:
-                return_document["updates"][0][key] = value
-        self.command_document = convert_to_camelcase(return_document)
+                 upsert=None, multi=None, collation=None, array_filters=None,
+                 hint=None, ordered=None, write_concern=None,
+                 bypass_document_validation=None, comment=None):
+        super().__init__(collection.name, collation)
+        self.command_document.update({
+            "updates":[{"q": filter, "u": update}]
+        })
+        if upsert is not None:
+            self.command_document["updates"][0]["upsert"] = upsert
```

Comment:
I suggest refactoring this to:
```
update_doc = {"q": filter, "u": update}
self.command_document["updates"] = [update_doc]
if upsert is not None:
   update_doc["upsert"] = upsert
...
```

```

Reply:
I don't see this change. Did it get lost?

Reply:
I refactored it incorrectly, it is done now.

Reply:
I still don't think this is complete. We can replace `self.command_document["updates"][0]` with `update_doc` now.

Reply:
Done.